### PR TITLE
timestamp validator

### PR DESCRIPTION
**Pull request by yisraelU**
ensures all members reachable from a service annotated with SimpleRestJson must have a timestamp format too

### CODE REVIEW

**Comment by lewisjkl** [comment]
Just pushed a commit with header fix. I think my auto-formatter changed the java file too, feel free to revert that.

On another note, I think adding a test to make sure that it has zero warnings when the service doesn't have `@simpleRestJson` would be good. Oh and also making sure that timestamps that ARE annotated with the timestampFormat trait raise no warnings.

File: modules/core/test/src/alloy/validation/SimpleRestJsonTimestampFormatValidatorSpec.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      ValidationEvent
+        .builder()
+        .id("SimpleRestJsonTimestamp")
+        .severity(Severity.WARNING)
+        .message(
+          "test#time: Timestamp shape test#time does not have a timestamp format trait "
+        )
+        .build()
+    )
+    assertEquals(result.size, 2)
```
Comment (lewisjkl) [inline modules/core/test/src/alloy/validation/SimpleRestJsonTimestampFormatValidatorSpec.scala:78]: Should update this assertion to use the expected list

File: modules/core/src/alloy/validation/SimpleRestJsonTimestampValidator.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ private Stream<ValidationEvent> validateTimestamp(MemberAndTarget shape) {
 	}
 
 	private ValidationEvent warn(Shape shape) {
-		return warning(shape, "Timestamp shape " + shape.getId() + " does not have a timestamp format trait");
+		return warning(shape, "A Timestamp shape  does not have a timestamp format trait");
```
Comment (daddykotex) [inline modules/core/src/alloy/validation/SimpleRestJsonTimestampValidator.java:76]:
```suggestion
		return warning(shape, "A Timestamp shape does not have a timestamp format trait");
```
minor double space

File: modules/core/test/src/alloy/validation/SimpleRestJsonTimestampFormatValidatorSpec.scala [hunk 3]
```diff
@@ -74,12 +74,24 @@ final class SimpleRestJsonTimestampFormatValidatorSpec extends munit.FunSuite {
...
         )
-        .build()
+        .build(),
+      ValidationEvent
+        .builder()
+        .id("SimpleRestJsonTimestamp")
+        .shape(member0)
+        .severity(Severity.WARNING)
+        .message(
+          "A Timestamp shape  does not have a timestamp format trait"
```
Comment (daddykotex) [inline modules/core/test/src/alloy/validation/SimpleRestJsonTimestampFormatValidatorSpec.scala:88]:
```suggestion
          "A Timestamp shape does not have a timestamp format trait"
```

File: modules/core/test/src/alloy/validation/SimpleRestJsonTimestampFormatValidatorSpec.scala [hunk 4]
```diff
@@ -74,12 +74,24 @@ final class SimpleRestJsonTimestampFormatValidatorSpec extends munit.FunSuite {
...
+        )
+        .build(),
+
     )
     assertEquals(result.size, 2)
```
Comment (daddykotex) [inline modules/core/test/src/alloy/validation/SimpleRestJsonTimestampFormatValidatorSpec.scala:93]: I think you should drop the size check, the only thing it does is hinder debugging when a failure occurs. `0 != 1` is much less helpful than `List(something) != List.empty`

wdyt?

Comment (daddykotex) [reply modules/core/test/src/alloy/validation/SimpleRestJsonTimestampFormatValidatorSpec.scala:93]: same for other checks below

**Review by daddykotex — APPROVED** [review]
👍

**Comment by daddykotex** [comment]
I'm not sure why it was all green but tests were still broken. Updated it in https://github.com/disneystreaming/alloy/pull/19
